privacyidea/app.py

```diff
@@ -218,18 +232,53 @@ def create_app(config_name="development",
         DEFAULT_LOGGING_CONFIG["loggers"]["privacyidea"]["level"] = level
         logging.config.dictConfig(DEFAULT_LOGGING_CONFIG)
 
-    babel = Babel(app)
-
-    @babel.localeselector
-    def get_locale():
-        return get_accepted_language(request)
-
     queue.register_app(app)
 
-    if init_hsm:
+    if initialize_hsm:
         with app.app_context():
             init_hsm()
 
+    if config_name == "testing":
+        # we are running the tests, create the database
+        with app.app_context():
+            db.create_all()
+
+    # check that we have a correct node_name -> UUID relation
+    with app.app_context():
+        # first check if we have a UUID in the config file which takes precedence
+        pi_uuid = app.config.get("PI_UUID")
+        if not pi_uuid:
+            # we try to get the unique installation id (See <https://0pointer.de/blog/projects/ids.html>)
+            try:
+                with open('/etc/machine-id', 'r') as f:
+                    pi_uuid = uuid.UUID(f.read().strip())
+            except Exception as e:  # pragma: no cover
+                logging.getLogger(__name__).info(f"Could not determine the machine id: {e}")
+            else:
+                # we generate a random UUID which will change on every startup
+                pi_uuid = uuid.uuid4()
+                logging.getLogger(__name__).warning("Generating a random UUID! "
+                                                    "If persisting the UUID fails, "
+                                                    "it will change on every application start")
+            app.config["PI_UUID"] = str(pi_uuid)
+            # safe UUID to config file
+            try:
+                with open(config_file, 'a') as f:  # pragma: no cover
+                    f.write(f"PI_UUID = \"{str(pi_uuid)}\"\n")
```

Comment:
It totally makes sense to save the uuid to a file, so that we do not need to run the maybe time consuming getting-uuid-from-machine-id at each app start.

However, I most probably would not recommend to make the config_file writeable. 
Could we save the uuid to a special uuid file? So that the privacyIDEA application only needs write access to this file?

Reply:
Since the `machine-id` is just a file, there shouldn't be much overhead. Anyway i have added the `PI_UUID_FILE` configuration which defaults to `/etc/privacyidea/uuid.txt`, where the UUID is saved when a new one is generated.
Done in c0c5f2d5e